### DISCUSSION TITLE
Module seems to not work in constrained language mode (files signed)

### DISCUSSION THREAD

**Discussion by ckuever** [HotCakeX/Harden-Windows-Security#272] [Q&A] [answered]
### Tools category

Harden Windows Security Module

### Does your system meet the requirements?

- [X] Yes, my system meets the requirements 👍

### Is your Windows installation genuine?

- [X] Yes, I am using genuine Windows installation. 💯

### Please explain the bug

Hi,

re-opening https://github.com/HotCakeX/Harden-Windows-Security/issues/243 with information

we did everything as you also nicely described

https://github.com/HotCakeX/Harden-Windows-Security/wiki/Script-Enforcement-and-PowerShell-Constrained-Language-Mode-in-WDAC-App-Control-Policies

However even if the modules could be loaded with no issues, there are no commands available

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/33624101/699b20a8-7d54-41b0-9296-5b82f0229906)

have you ever tried your nice modules with script enforcement and in constrained language mode? Or will this simply not work?

Thank you.

**Comment by HotCakeX** [comment]
Hi,
The modules here use advanced features that aren't allowed in constrained language mode even when signed (which isn't what i had expected but that's what i experienced), maybe the PowerShell team can shed some light on it later.

That's why I personally don't recommended script enforcement unless the device isn't going to use PowerShell except for some simple built-in commands.

---

since it's not a bug, at least not one that i can fix, i'm gonna move it to discussions, i will update you as soon as i have new info on this 🙂

**Comment by ckuever** [comment, marked as answer]
all fine and can be closed :-) we expected that and thanks for confirmation

**Reply by HotCakeX** [reply]
No problem, i will keep you posted if i have more info about this subject.
